### ISSUE TITLE
Strange error

### ISSUE THREAD

**Issue by iantearle** [k0sukey/alloy.widget.iconicfont#2]
I can include the widget via JS, but for some reason not XML, I am getting the following error, have you have seen this before?

``` javascript
[INFO]    created:    "Resources/alloy/widgets/be.k0suke.ssforecast/controllers/widget.js"
[DEBUG]   map:        "build/map/Resources/alloy/widgets/be.k0suke.ssforecast/controllers/widget.js.map"
[INFO]  [index.xml] view processing...
[INFO]    style:      "index.tss"
[INFO]    view:       "index.xml"
[INFO]    controller: "index.js"
[TRACE] - Processing "builtins" module...
[TRACE] - Processing "optimizer" module...
[TRACE] - Processing "compress" module...
[INFO]    created:    "Resources/alloy/controllers/index.js"
[DEBUG]   map:        "build/map/Resources/alloy/controllers/index.js.map"
[INFO]  [weather.xml] view processing...
[INFO]    style:      "weather.tss"
[INFO]    view:       "weather.xml"
[ERROR] [ERROR] widget "be.k0suke.ssforecast" view "widget" does not exist.
[ERROR] [ERROR] The following paths were inspected:
[ERROR] [ERROR] /Users/iantearle/Titanium/SomethingBlew/app/widgets/be.k0suke.ssforecast/views/ios/widget
[ERROR] [ERROR] /Users/iantearle/Titanium/SomethingBlew/app/widgets/be.k0suke.ssforecast/views/widget
[ERROR] [ERROR] /usr/local/lib/node_modules/alloy/widgets/be.k0suke.ssforecast/views/ios/widget
[ERROR] [ERROR] /usr/local/lib/node_modules/alloy/widgets/be.k0suke.ssforecast/views/widget
[ERROR] Alloy compiler failed
```

I'm wondering if the latest alloy has changed something?

**Comment by iantearle** [comment]
Right, have whittled it down to this.

If I change the name to: be.k0suke.forecast it works. It seems that be.k0suke.ssforecast has too many characters for alloy XML to deal with. Works fine from inclusion from JS.

This works:

``` javascript
var sunrise = Alloy.createWidget("be.k0suke.ssforecast", "widget", {
        id: "ToolbarIcon",
        icon: 'sunrise',
        color: '#d6edf6',
        font: {fontSize: 24},
        height: Ti.UI.SIZE,
        width: Ti.UI.SIZE,
        top: 0, left: 0,
        textAlign: 'center',
        backgroundSelectedColor: '#ffffff'
    }).getView();

    $.sunriseIcon.add(sunrise);
```

This does not:

< Widget class="icon" id="xtraIcon" icon="sunset" src="be.k0suke.ssforecast"/ >

Where as this does:

< Widget class="icon" id="xtraIcon" icon="sunset" src="be.k0suke.forecast"/ >
- GitHub dont seem to like XML.

**Comment by tonylukasavage** [comment]
I'd be shocked if "too many characters" was actually the root of the problem, since we definitely have widgets with longer names than that being used in XML. What's the full path to the folder that your widget is contained in?

**Comment by iantearle** [comment]
That'd be "/Users/iantearle/Titanium/SomethingBlew/app/widgets/be.k0suke.ssforecast/views/widget.xml" - I tried changing just the name removing characters one at a time from then end and got the same error with anything over be.k0suke.ssforeca (18 chars) and, yes I was changing the name in all necessary locations. In my testing this way I finally deleted the widget from the Resources folder, and remarkably it works. So am still baffled as to why it was returning that error. 
Shall close this here now and report any further occurances. Thanks for taking a look at this Tony.
